src/main/java/com/lgtm/easymoney/controllers/UserController.java

```diff
@@ -1,31 +1,30 @@
 package com.lgtm.easymoney.controllers;
 
-import com.lgtm.easymoney.models.User;
 import com.lgtm.easymoney.payload.BalanceReq;
 import com.lgtm.easymoney.payload.BalanceRsp;
-import com.lgtm.easymoney.payload.ErrorRsp;
 import com.lgtm.easymoney.services.UserService;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import javax.validation.Valid;
-import java.math.BigDecimal;
-import java.util.ArrayList;
-import java.util.Arrays;
 
 @RestController
 @RequestMapping("/user")
 public class UserController {
+    private final UserService userService;
+
     @Autowired
-    private UserService userService;
+    public UserController(UserService userService) {
```

Comment:
curious: why we can't simply use the original one?
` @Autowired
    private UserService userService;`

Reply:
Intellij showed a warning like direct injection is not recommended